frontend/src/main/scala/bloop/engine/caches/SemanticDBCache.scala

```diff
@@ -0,0 +1,72 @@
+package bloop.engine.caches
+
+import bloop.logging.Logger
+import coursier.core.Repository
+import bloop.io.AbsolutePath
+import bloop.io.Paths
+import bloop.DependencyResolution
+import scala.util.Try
+import java.nio.file.Files
+import xsbti.ComponentProvider
+import xsbti.GlobalLock
+import java.io.File
+import java.util.concurrent.Callable
+import sbt.internal.inc.bloop.ZincInternals
+import sbt.internal.inc.ZincComponentManager
+import sbt.internal.inc.IfMissing
+import scala.util.Failure
+import scala.util.Success
+import bloop.ComponentLock
+import bloop.SemanticDBCacheLock
+import java.nio.file.Path
+import coursier.error.CoursierError
+
+object SemanticDBCache {
+  @volatile private var latestResolvedSemanticDB: Path = null
+  def fetchPlugin(
+      scalaVersion: String,
+      version: String,
+      logger: Logger
+  ): Either[String, AbsolutePath] = {
+    val organization = "org.scalameta"
+    val module = s"semanticdb-scalac_$scalaVersion"
+    val semanticDBId = s"$organization.$module.$version"
+    val provider = ZincInternals.getComponentProvider(Paths.getCacheDirectory("semanticdb"))
+    val manager = new ZincComponentManager(SemanticDBCacheLock, provider, secondaryCacheDir = None)
+
+    def attemptResolution: Either[String, AbsolutePath] = {
+      import bloop.engine.ExecutionContext.ioScheduler
+      DependencyResolution.resolveWithErrors(organization, module, version, logger)(ioScheduler) match {
+        case Left(error) => Left(error.getMessage())
+        case Right(paths) =>
+          paths.find(_.syntax.contains("semanticdb-scalac")) match {
+            case Some(pluginPath) => Right(pluginPath)
+            case None =>
+              Left(
+                s"Missing semanticdb plugin in resolved jars ${paths.map(_.syntax).mkString(",")}"
+              )
+          }
+      }
+    }
+
+    if (version == "latest.release") {
```

Comment:
Mostly left this as you made it @tgodzik with the exception that if the version is `latest.release` we only resolve it once per instance of the bloop server.